### ISSUE TITLE
Feature Request - New Width Classes

### ISSUE THREAD

**Issue by mrlife** [twbs/bootstrap#31289]
`width: fit-content` -> `w-fit` would be great

Other options:

`width: min-content` -> `w-min`
`width: max-content` -> `w-max`

Could add the same for height, but not sure it's as useful.

**Comment by ffoodd** [comment]
Could you provide any use cases for those—and also, would there be any sense to add those values to `min-width` and `max-width` too?

**Comment by ffoodd** [comment]
Indeed, it helps :) 

I just came across another use case for `min-content` and `max-content`: [a codepen by Sven Wolfermann](https://codepen.io/maddesigns/pen/YNXvMz), shared by Stefan Judis in a dedicated TIL post](https://www.stefanjudis.com/today-i-learned/a-use-case-for-css-min-content-and-max-content/).

**Comment by juhas96** [comment]
Hey I want to contribute on this issue

**Comment by septatrix** [comment]
Also `.w-max-{sm,md,lg}` classes (or a similar name) would be useful which set `max-width` to the respective values from `$container-max-widths`. Often I see the use case that the largest width from `.container` is too wide for many layouts and looks bad. This is also manually done in some examples from the gallery. For example when removing this manual `max-width` from the checkout form and removing the second column it loses its clean look and becomes hard to parse. A `.w-max-md` would work wonders here:

![image](https://user-images.githubusercontent.com/24257556/182862038-307eddff-d2f6-424c-bcaf-63941e5afb12.png)

**Comment by mdo** [comment]
Closing per https://github.com/twbs/bootstrap/pull/31943#issuecomment-1265830492.
